Packages/com.unity.inputsystem/Documentation~/UISupport.md

```diff
@@ -142,3 +150,27 @@ Note that the resulting [Mouse](../api/UnityEngine.InputSystem.Mouse.html) input
 
 >[!NOTE]
 >Do not set up gamepads and joysticks for [navigation input](#navigation-type-input) while using VirtualMouseInput. If both VirtualMouseInput and navigation are configured, input is triggered twice: once via the pointer input path, and once via the navigation input path. If you encounter problems such as where buttons are pressed twice, this is likely the problem.
+
+# UI Toolkit support
+
+As of Unity 2021.2, [UI Toolkit](https://docs.unity3d.com/Manual/UIElements.html) is supported as an alternative to the [Unity UI](https://docs.unity3d.com/Manual/com.unity.ugui.html) system for implementing UIs in players.
+
+Input support for both [Unity UI](https://docs.unity3d.com/Manual/com.unity.ugui.html) and [UI Toolkit](https://docs.unity3d.com/Manual/UIElements.html) is based on the same [`EventSystem`](https://docs.unity3d.com/Packages/com.unity.ugui@1.0/manual/EventSystem.html) and [`BaseInputModule`](https://docs.unity3d.com/Packages/com.unity.ugui@1.0/manual/InputModules.html) subsystem. In other words, the same input setup based on [`InputSystemUIInputModule`](#inputsystemuiinputmodule-component) supports input in either UI solution and nothing extra needs to be done.
+
+Internally, UI Toolkit will install an event listener in the form of the `PanelEventHandler` component which intercepts events sent by `InputSystemUIInputModule` and translates them into UI Toolkit-specific events that are then routed into the visual tree. Note that if you employ `EventSystem.SetUITookitEventSystemOverride`, this default mechanism is bypassed.
+
+>[!NOTE]
+>XR ([tracked-type input](#tracked-type input)) is not yet supported in combination with UI Toolkit. This means that devices such as VR controllers cannot yet be used to operate interfaces created with UI Toolkit.
+
+There are some additional things worth noting:
+
+* Raycasting is handled internally by UI Toolkit. No separate raycaster component is needed like for uGUI. This also means that [`TrackedDeviceRaycaster`](../api/UnityEngine.InputSystem.UI.TrackedDeviceRaycaster.html) will not work together with UI Toolkit.
```

Comment:
In fact, we're using a `PanelRaycaster` on the same object that has the `PanelEventHandler`. That PanelRaycaster inherits from BaseRaycaster, which means it will be taken into consideration by the UGUI raycasting process, and will make it so that EventSystem.current.IsPointerOverGameObject() will return true if the pointer is over a UITK panel. I'm not sure if this contradicts what you wrote here, just making sure you knew about this.

Reply:
Checking that paragraph, I'm thinking it sufficiently aligns with the mechanism you're describing. Basically, as far as the user is concerned, the outcome is "I don't need to worry about raycasting. Plus TrackedDeviceRaycaster isn't working.". I'll go with the current wording for now.